cwl/src/main/scala/cwl/CommandLineTool.scala

```diff
@@ -40,23 +43,55 @@ case class CommandLineTool private(
                                    temporaryFailCodes: Option[Array[Int]],
                                    permanentFailCodes: Option[Array[Int]]) {
 
-  def womExecutable(inputFile: Option[String] = None): Checked[Executable] =
-    CwlExecutableValidation.buildWomExecutable(ExecutableTaskDefinition.tryApply(taskDefinition).toEither, inputFile)
+  /** Builds an `Executable` directly from a `CommandLineTool` CWL with no parent workflow. */
+  def womExecutable(validator: Requirement => ErrorOr[Requirement], inputFile: Option[String] = None): Checked[Executable] = {
```

Comment:
This is the current shape for a `Requirement` validator function. It takes a `Requirement` and opines on its acceptability, returning the same `Requirement` in a `Valid` or an explanation for why it was unacceptable in an `Invalid[NonEmptyList[String]]`. Unacceptable `Requirement`s need to be handled differently depending on whether they are true `requirements` or merely `hints`. Any unsuccessful validations should be returned to the caller to address the error handling that's beyond the scope of this code.